framework/src/utils/KDTree.C

```diff
@@ -48,3 +48,26 @@ KDTree::neighborSearch(Point & query_point,
   return_index.resize(n_result);
   return_dist_sqr.resize(n_result);
 }
+
+void
+KDTree::radiusSearch(Point & query_point,
+                     Real radius,
+                     std::vector<std::size_t> & return_index,
+                     std::vector<Real> & return_dist_sqr)
+{
+  std::vector<std::pair<std::size_t, Real>> indices_dist;
```

Comment:
Any reason you don't make the signature 
```
KDTree::radiusSearch(Point & query_point,
                     Real radius,
                     std::vector<std::pair<std::size_t, Real>> indices_dist);
```
and save the copy operation?

Reply:
Oh, because the other function make the same mistake... ;-)

Reply:
`neighborSearch` and `radiusSearch` can be changed to do it right....

Reply:
The inconsistency comes from the `nanoflann` interface. `radiusSearch` wants a `std::vector<std::pair<unsigned int, Real>>`, while `knnSearch` wants two separate vectors.